### PR TITLE
chore!(v0): update v0 for the upcoming ckb 2023 hardfork

### PR DESCRIPTION
**Pull request by blckngm**
Updated rust, ckb deps, ckb-vm, ckb-sdk and ckb-cli. The ckb builtin indexer is required (and the standalone indexer is no longer supported).

- [x] A new image would need to be built. It should have a newer ckb-cli.
   - [x] test: use the new image to run a Godwoken v0 readonly-node and sync to TIP to make sure the new image is OK
- [x] Kicker will also need some small updates.

### CODE REVIEW

**Comment by Flouse** [comment]
The ckb builtin indexer is required and the standalone ckb-indexer is no longer supported in the next version of Godwoken v0.
cc @Simon-Tl

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:20.04
+
+# Install curl.
+RUN apt-get update \
+ && apt-get install -y curl \
+ && rm -rf /var/lib/apt/lists/*
+
+# Copy scripts
+COPY --from=ghcr.io/zeroqn/godwoken-prebuilds:dev-feat-fast-withdrawal-to-v1.1 /scripts /scripts
```
Comment (Flouse) [inline Dockerfile]: # Checksum

## image: ghcr.io/zeroqn/godwoken-prebuilds:dev-feat-fast-withdrawal-to-v1.1
```bash
docker run -it --rm ghcr.io/zeroqn/godwoken-prebuilds:dev-feat-fast-withdrawal-to-v1.1  bash
find /scripts -type f -exec sha1sum {} \;

# Results
ead44e1b236bbae0f92a41906f6521924a70518b  /scripts/clerkb/state
7be1fdabefdf35121254400577631759217772af  /scripts/clerkb/poa
da39a3ee5e6b4b0d3255bfef95601890afd80709  /scripts/godwoken-scripts/.keep
f15863d3cfb3d7a64d1d2017692e983c88e0d4b7  /scripts/godwoken-scripts/meta-contract-validator
e21a690b7fbd604227e25185653e05518dbe7ecf  /scripts/godwoken-scripts/sudt-validator
24bd3cb7b53798997625be57c2313961c1cb2217  /scripts/godwoken-scripts/sudt-generator
8810738bcff762f958bbb2f4d961c5a72f25050d  /scripts/godwoken-scripts/meta-contract-generator
acdbf1bdf3932d2df6b06aba288ad4b9751e0f79  /scripts/godwoken-scripts/tron-account-lock
32599cb35eaf7f7b804a3c7596ff96073aeecca4  /scripts/godwoken-scripts/state-validator
248c7d2d85749d163423f3477b12231de1a9b3cd  /scripts/godwoken-scripts/always-success
75d72d20d75c4bd4ddff816cf0fad24f36019468  /scripts/godwoken-scripts/stake-lock
2488bcf5240ebeeff537b60a63cd88a628777bbe  /scripts/godwoken-scripts/eth-account-lock
456c10df920da772791b43bdf2c4a8c1024655eb  /scripts/godwoken-scripts/challenge-lock
10a79886363a68c7fae18b712ed9a9d7d9da3b6c  /scripts/godwoken-scripts/deposit-lock
ca33d354ec2cb7cbc4a572c5c63729e2c6a57b18  /scripts/godwoken-scripts/withdrawal-lock
cdb45e361157c4446b1d9bc1b63f2895dcfdc453  /scripts/godwoken-scripts/custodian-lock
acbd9e313fa7346b71611a9bdfe34153781d1f92  /scripts/godwoken-polyjuice/validator
26680e94756a697c6e09553df8332eb2ba86bf5b  /scripts/godwoken-polyjuice/validator_log
5f73c1fc4a8e71a3986444b191ce61e71383699e  /scripts/godwoken-polyjuice/validator.debug
486f05946a2714f49453420a8b20f46366a514e3  /scripts/godwoken-polyjuice/validator_log.debug
52a7a9394dee98383cf40ee6fdba64a085758272  /scripts/godwoken-polyjuice/generator
94fa77fed621f4c7dc1247b9124d608ce8c5f2d5  /scripts/godwoken-polyjuice/generator.asm
67891ce0b3633fde5fd59cd1a0aaefc7e9e5c290  /scripts/godwoken-polyjuice/generator_log
05b58df914d3c8c98e4857f558bea10882bb4283  /scripts/godwoken-polyjuice/generator_log.asm
52a7a9394dee98383cf40ee6fdba64a085758272  /scripts/godwoken-polyjuice/generator.aot
67891ce0b3633fde5fd59cd1a0aaefc7e9e5c290  /scripts/godwoken-polyjuice/generator_log.aot
d02617b266f71d88a3a36123e72508ced56fa725  /scripts/godwoken-polyjuice/generator.debug
24cc9c26abcafd2ae59be5710b20e2e6d9b44ed4  /scripts/godwoken-polyjuice/generator_log.debug
```

## image: ghcr.io/nervosnetwork/godwoken-prebuilds:v0.10.7
```bash
docker run -it --rm ghcr.io/nervosnetwork/godwoken-prebuilds:v0.10.7  bash
find /scripts -type f -exec sha1sum {} \;

# Results
ead44e1b236bbae0f92a41906f6521924a70518b  /scripts/clerkb/state
7be1fdabefdf35121254400577631759217772af  /scripts/clerkb/poa
da39a3ee5e6b4b0d3255bfef95601890afd80709  /scripts/godwoken-scripts/.keep
f15863d3cfb3d7a64d1d2017692e983c88e0d4b7  /scripts/godwoken-scripts/meta-contract-validator
e21a690b7fbd604227e25185653e05518dbe7ecf  /scripts/godwoken-scripts/sudt-validator
24bd3cb7b53798997625be57c2313961c1cb2217  /scripts/godwoken-scripts/sudt-generator
8810738bcff762f958bbb2f4d961c5a72f25050d  /scripts/godwoken-scripts/meta-contract-generator
acdbf1bdf3932d2df6b06aba288ad4b9751e0f79  /scripts/godwoken-scripts/tron-account-lock
32599cb35eaf7f7b804a3c7596ff96073aeecca4  /scripts/godwoken-scripts/state-validator
248c7d2d85749d163423f3477b12231de1a9b3cd  /scripts/godwoken-scripts/always-success
75d72d20d75c4bd4ddff816cf0fad24f36019468  /scripts/godwoken-scripts/stake-lock
2488bcf5240ebeeff537b60a63cd88a628777bbe  /scripts/godwoken-scripts/eth-account-lock
456c10df920da772791b43bdf2c4a8c1024655eb  /scripts/godwoken-scripts/challenge-lock
10a79886363a68c7fae18b712ed9a9d7d9da3b6c  /scripts/godwoken-scripts/deposit-lock
ca33d354ec2cb7cbc4a572c5c63729e2c6a57b18  /scripts/godwoken-scripts/withdrawal-lock
cdb45e361157c4446b1d9bc1b63f2895dcfdc453  /scripts/godwoken-scripts/custodian-lock
327c4189136f46146b006ad26c4378d9b1b0697c  /scripts/godwoken-polyjuice/validator
14d1bc319771854d0c8a58b6cf6fbff609891589  /scripts/godwoken-polyjuice/validator_log
83b34931441c034e8508859882b89e2eb1cbc978  /scripts/godwoken-polyjuice/validator.debug
8ac502f079414bc3511210004aef019eb21cf222  /scripts/godwoken-polyjuice/validator_log.debug
2c6eec6471bc43d7fbef4cea996ccb45bbfb1128  /scripts/godwoken-polyjuice/generator
d357299a3cf05a38083c515588a92396022b4e33  /scripts/godwoken-polyjuice/generator.asm
2052c100212fa712b46d8481bc18535414aba8d1  /scripts/godwoken-polyjuice/generator_log
3c893963783bc1f60b992aa6e898fc8c024d5fa7  /scripts/godwoken-polyjuice/generator_log.asm
2c6eec6471bc43d7fbef4cea996ccb45bbfb1128  /scripts/godwoken-polyjuice/generator.aot
2052c100212fa712b46d8481bc18535414aba8d1  /scripts/godwoken-polyjuice/generator_log.aot
c9377998fd63d45f00d4421f4d599e7eb48362c8  /scripts/godwoken-polyjuice/generator.debug
c762fa8f4a67b1d10b603c0b2e5cc7d8c317a03f  /scripts/godwoken-polyjuice/generator_log.debug

```

Comment (Flouse) [inline Dockerfile]:
```suggestion
COPY --from=ghcr.io/nervosnetwork/godwoken-prebuilds:v0.10.7 /scripts /scripts
```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Flouse) [reply Dockerfile]: @jjyr Please help us to double check.

**Comment by blckngm** [comment]
I've tested this with https://github.com/godwokenrises/godwoken-kicker/commit/c65db3158db0995dcec46a09170ec24b38e432fc and v0 works fine.

Now we just need to add a github workflow to build and push the image.

**Comment by Flouse** [comment]
please update the image used in https://github.com/godwokenrises/godwoken/blob/0.10-rc/devtools/fetch-binaries.sh too.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:20.04
```
Comment (Flouse) [inline Dockerfile]:
```suggestion
# For the previous image building workflow, see https://github.com/godwokenrises/godwoken-docker-prebuilds/blob/0.10-rc

FROM ubuntu:20.04
```

Comment (blckngm) [reply Dockerfile]: We'll still use godwoken-docker-prebuilds for v0.

File: crates/jsonrpc-types/src/blockchain.rs [hunk 3]
```diff
@@ -28,6 +24,7 @@ impl From<ScriptHashType> for packed::Byte {
             ScriptHashType::Data => packed::Byte::new(0),
             ScriptHashType::Type => packed::Byte::new(1),
             ScriptHashType::Data1 => packed::Byte::new(2),
+            ScriptHashType::Data2 => packed::Byte::new(3),
```
Comment (blckngm) [inline crates/jsonrpc-types/src/blockchain.rs]:
```suggestion
            ScriptHashType::Data2 => packed::Byte::new(4),
```

Comment (jjyr) [reply crates/jsonrpc-types/src/blockchain.rs]: https://github.com/nervosnetwork/ckb/blob/49b05e102d620c6072d384b882ef7a6519bf11cb/util/gen-types/src/core.rs#L65

**Review by jjyr — APPROVED** [review]

**Comment by jjyr** [comment]
Please fix CI @blckngm

**Comment by blckngm** [comment]
We should update to ckb 0.111.0 and ckb-sdk 3.0. Will update later.

**Review by Flouse — APPROVED** [review]

**Pull request merged**